src/ring_theory/noetherian.lean

```diff
@@ -69,6 +70,20 @@ theorem fg_def {N : submodule R M} :
   exact ⟨t, rfl⟩
 end⟩
 
+lemma fg_iff_add_submonoid_fg (P : submodule ℕ M) :
+  P.fg ↔ P.to_add_submonoid.fg :=
+begin
+  split,
+  { rintro ⟨S, hS⟩,
+    refine ⟨S, _⟩,
+    rw [← submodule.span_nat_eq_add_submonoid_closure],
+    simp only [hS, submodule.top_to_add_submonoid] },
```

Comment:
Did you already fix up this proof? I'm a little surprised that `top_to_add_submonoid` is needed here, since `top` isn't anywhere in the statement or goal.

Untested because I don't have a cache, but
```suggestion
    rw [← span_nat_eq_add_submonoid_closure, hS], },
```
You're in the `submodule` namespace here so you can trim `submodule.` from the proof in a few places.

Reply:
Golfed!